GitUI/CommandsDialogs/FormDeleteBranch.cs

```diff
@@ -78,23 +68,35 @@ private void Delete_Click(object sender, EventArgs e)
                 return;
             }
 
-            if (!AppSettings.DontConfirmDeleteUnmergedBranch)
+            bool areAllInReflog = selectedBranches.All(b => _reflogHashes.Contains(b.ObjectId.ToString()));
+
+            // Detect if commits will be dangling (i.e. no remaining local refs left handling commit)
+            string[] deletedCandidates = selectedBranches.Select(b => b.Name).ToArray();
+            bool atLeastOneHeadCommitWillBeDangling = false;
+
+            foreach (IGitRef selectedBranch in selectedBranches)
             {
-                Validates.NotNull(_mergedBranches);
+                atLeastOneHeadCommitWillBeDangling = !GetAllBranchesWhichContainGivenCommit(selectedBranch.ObjectId)
+                    .Any(b2 => !deletedCandidates.Contains(b2));
```

Comment:
This is another O(n^2) repeated in a loop also making git calls...

Reply:
Here I admit that the performance could be a little more "concerning".
But for me it's still acceptable for the reasons:
1. As said before, more of the times, you will delete only 1 branch and so, in this specific case, you will see **NO** delay because check for the selected commit has been cached (because already done in the info panel). **And in all these cases, the user will gain time because the goal of this check is to prevent the user spending time to answer a frustrating and useless question**

3. If you select multiple branches, it could indeed take some time but:
    * as said before, it takes the user much more time to select the branches than do the check
    * check is threaded and started immediately after the branch selection so some time is "hidden" the time the user verify the selection or click "Delete"
    * if performance really matter on this form, we would have already seen that loading form if there are a lot of branches is taking much more time **everytimes** than this check (fix will follow)
    * it worth the wait because the check is of much more value than the existing one because if you have no question, you could be 100% sure that you won't loose something. And if you have the warning question, it saves your ass of minutes or probably hours or days if you do a mistake (what a gain of time!!)

Example on a repo with 4k branches and to delete 10 branches: 300ms of total check.
Example on a repo with 20k (!!) branches and to delete:
   * 10 branches relatively recent commits: 300ms of total check
   * 10 branches on the oldest commits (worst of the worst scenario): 15s (slow but not silly --approximately the time to open the popup at the moment--)
 
So overall, I would say acceptable....